src/System.Private.Interpreter/src/Internal/Runtime/Interpreter/ILImporter.Interpreter.cs

```diff
@@ -239,22 +255,1021 @@ private void ImportLoadFloat(double value)
+                case WellKnownType.SByte:
+                    {
+                        sbyte result = default(sbyte);
+                        if (stackItem.Kind == StackValueKind.Int32)
+                        {
+                            if (unsigned)
+                            {
+                                uint value = (uint)stackItem.AsInt32();
+                                result = checkOverflow ? Convert.ToSByte(value) : (sbyte)value;
+                            }
+                            else
+                            {
+                                int value = stackItem.AsInt32();
+                                result = checkOverflow ? Convert.ToSByte(value) : (sbyte)value;
+                            }
+                        }
+                        else if (stackItem.Kind == StackValueKind.Int64)
+                        {
+                            if (unsigned)
+                            {
+                                ulong value = (ulong)stackItem.AsInt64();
+                                result = checkOverflow ? Convert.ToSByte(value) : (sbyte)value;
+                            }
+                            else
+                            {
+                                long value = stackItem.AsInt64();
+                                result = checkOverflow ? Convert.ToSByte(value) : (sbyte)value;
+                            }
+                        }
+                        else if (stackItem.Kind == StackValueKind.Float)
+                        {
+                            double value = stackItem.AsDouble();
+                            result = checkOverflow ? Convert.ToSByte(value) : (sbyte)value;
+                        }
+                        else if (stackItem.Kind == StackValueKind.NativeInt)
+                        {
+                            if (unsigned)
+                            {
+                                UIntPtr value = (UIntPtr)stackItem.AsNativeInt().ToPointer();
+                                result = checkOverflow ? Convert.ToSByte(value.ToUInt64()) : (sbyte)value;
+                            }
+                            else
+                            {
+                                IntPtr value = stackItem.AsNativeInt();
+                                result = checkOverflow ? Convert.ToSByte(value.ToInt64()) : (sbyte)value;
+                            }
+                        }
+                        else
+                        {
+                            // TODO: Figure out how the CLR converts an object to an integer representation
+                        }
+
+                        _interpreter.EvaluationStack.Push(StackItem.FromInt32(result));
+                        break;
+                    }
+                case WellKnownType.Byte:
+                    {
+                        byte result = default(byte);
+                        if (stackItem.Kind == StackValueKind.Int32)
+                        {
```

Comment:
There is a lot of duplicated code. This should compute the stack value first in one switch or sequence of ifs; and then convert that value in a second switch.

Reply:
I'm confused on how this will work. If I compute the stack value separately I still can't assign them to a single variable of a single type. As a result, I'll still need if checks within the conversion switch to be sure of the type of the value

Reply:
You are right. There is not much to save in this case.